index.js

```diff
@@ -325,6 +325,20 @@ function iterableEqual(leftHandOperand, rightHandOperand, options) {
       return false;
     }
   }
+
+  var leftHandProps = Object.keys(leftHandOperand);
+  var rightHandProps = Object.keys(rightHandOperand);
+  if (leftHandProps.length !== rightHandProps.length) {
+    return false;
+  }
+
+  for (var prop in rightHandProps) {
```

Comment:
I'll try this and run the benchmarks to see if it's worth it, but I do agree it would be a lot more clean.